common/ledger/blkstorage/snapshot_test.go

```diff
@@ -0,0 +1,465 @@
+		blocksDetailsBeforeSnapshot,
+		blocksBeforeSnapshot,
+	)
+	verifyQueriesOnBlocksAddedAfterBootstrapping(t,
+		bootstrappedBlockStore,
+		expectedBCInfo,
+		blocksDetailsAfterSnapshot,
+		blocksAfterSnapshot,
+	)
+
+	// verify that bootstrapped blockstore itself can export TXIDs
+	anotherSnapshotDir := filepath.Join(testPath, "anotherSnapshot")
+	require.NoError(t, os.Mkdir(anotherSnapshotDir, 0755))
+	fileHashes, err := bootstrappedBlockStore.ExportTxIds(anotherSnapshotDir, testNewHashFunc)
+	require.NoError(t, err)
+	expectedTxIDs := []string{}
+	for _, b := range append(blocksDetailsBeforeSnapshot, blocksDetailsAfterSnapshot...) {
+		expectedTxIDs = append(expectedTxIDs, b.txIDs...)
+	}
+	sort.Slice(expectedTxIDs, func(i, j int) bool {
+		ith := expectedTxIDs[i]
+		jth := expectedTxIDs[j]
+		if len(ith) == len(jth) {
+			return ith < jth
+		}
+		return len(ith) < len(jth)
+	})
+	verifyExportedTxIDs(t, anotherSnapshotDir, fileHashes, expectedTxIDs...)
+
+	// verify that the bootstrapped blockstore can sync up indexes
+	toAddWithoutIndexupdates := []*testBlockDetails{
+		{
+			txIDs: []string{"txid_syncup_index_1", "txid_syncup_index_2"},
+		},
+		{
+			txIDs: []string{"txid_syncup_index_3", "txid_syncup_index_4"},
+		},
+	}
+
+	blocks := []*common.Block{}
+	blkfileMgr := bootstrappedBlockStore.fileMgr
+	originalIndexDB := blkfileMgr.index.db
+
+	// redirect index writes to some random place and add two blocks and then set the original index back
+	bootstrappedBlockStore.fileMgr.index.db = env.provider.leveldbProvider.GetDBHandle(filepath.Join(testPath, "someRandomPlace"))
+	for _, blockDetails := range toAddWithoutIndexupdates {
+		block := generateNextTestBlock(blocksGenerator, blockDetails)
+		blocks = append(blocks, block)
+		require.NoError(t, blkfileMgr.addBlock(block))
+	}
+	blkfileMgr.index.db = originalIndexDB
+
+	// before, we test for index sync-up, verify that the last set of blocks not indexed in the original index
+	for _, block := range blocks {
+		_, err := blkfileMgr.retrieveBlockByNumber(block.Header.Number)
+		assert.Exactly(t, ErrNotFoundInIndex, err)
+	}
+
+	// close and open should sync-up the index
+	env.provider.Close()
+	env = newTestEnv(t, NewConf(testPath, 0))
+	bootstrappedBlockStore, err = env.provider.Open("bootstrappedLedger")
+	require.NoError(t, err)
+
+	expectedBCInfo = &common.BlockchainInfo{
+		Height:            blocks[1].Header.Number + 1,
+		CurrentBlockHash:  protoutil.BlockHeaderHash(blocks[1].Header),
+		PreviousBlockHash: blocks[1].Header.PreviousHash,
+	}
+
+	verifyQueriesOnBlocksAddedAfterBootstrapping(
+		t,
+		bootstrappedBlockStore,
+		expectedBCInfo,
+		toAddWithoutIndexupdates,
+		blocks,
+	)
+
+	// verify that the bootstrapped blockstore returns error if indexes are deleted
+	env.provider.Close()
+	conf := NewConf(testPath, 0)
+	require.NoError(t, os.RemoveAll(conf.getIndexDir()))
+	env = newTestEnv(t, conf)
+	bootstrappedBlockStore, err = env.provider.Open("bootstrappedLedger")
```

Comment:
Nit

`_, err = env.provider.Open("bootstrappedLedger")` or an assert on `bootstrappedBlockStore`.

Reply:
sure, will fix this as well, when I split the test